iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java

```diff
@@ -91,6 +98,33 @@ private boolean checkOrders(int side, VoltageLevel voltageLevel, Reporter report
         return true;
     }
 
+    private boolean checkOrderValue(int side, BusbarSection busbarSection) {
+        VoltageLevel voltageLevel = busbarSection.getTerminal().getVoltageLevel();
+        Integer positionOrder = getPositionOrder(side);
+        BusbarSectionPosition positionExtension = busbarSection.getExtension(BusbarSectionPosition.class);
+        if (positionExtension != null) {
+            Optional<Range<Integer>> unusedOrderPositionsAfter = getUnusedOrderPositionsAfter(busbarSection);
+            Optional<Range<Integer>> unusedOrderPositionsBefore = getUnusedOrderPositionsBefore(busbarSection);
```

Comment:
Here you get a range whereas you only need one of the two values. And calling the two methods makes you calculating the map of position twice. We could avoid this by calculating directly the desired values.